src/spikeinterface/curation/tests/test_train_manual_curation.py

```diff
@@ -0,0 +1,206 @@
+    temp_file_path = make_temp_training_csv()
+
+    # Load and preprocess the data from the temporary CSV file
+    trainer.load_and_preprocess_csv([temp_file_path])
+
+    # Assert that the data is loaded and preprocessed correctly
+    assert trainer.X is not None
+    assert trainer.y is not None
+    assert trainer.testing_metrics is not None
+
+
+def test_apply_scaling_imputation(trainer):
+
+    imputation_strategy = "knn"
+    scaling_technique = "standard_scaler"
+    X_train = np.array([[1, 2, 3], [4, 5, 6]])
+    X_val = np.array([[7, 8, 9], [10, 11, 12]])
+    y_train = np.array([0, 1])
+    y_val = np.array([2, 3])
+    X_train_scaled, X_val_scaled, y_train, y_val, imputer, scaler = trainer.apply_scaling_imputation(
+        imputation_strategy, scaling_technique, X_train, X_val, y_train, y_val
+    )
+    assert X_train_scaled is not None
+    assert X_val_scaled is not None
+    assert y_train is not None
+    assert y_val is not None
+    assert imputer is not None
+    assert scaler is not None
+
+
+def test_get_classifier_search_space(trainer):
+
+    classifier = "LogisticRegression"
+    model, param_space = trainer.get_classifier_search_space(classifier)
+    assert model is not None
+    assert isinstance(param_space, dict)
+
+
+def test_get_custom_classifier_search_space():
+    classifier = {
+        "LogisticRegression": {
+            "C": [0.001, 8.0],
+            "solver": ["newton-cg", "lbfgs", "liblinear", "sag", "saga"],
+            "max_iter": [100, 400],
+        }
+    }
+    trainer = CurationModelTrainer(classifiers=classifier, labels=[[0, 1, 0, 1, 0, 1, 0, 1, 0, 1]])
+
+    model, param_space = trainer.get_classifier_search_space(list(classifier.keys())[0])
+    assert model is not None
+    assert param_space == classifier["LogisticRegression"]
+
+
+def test_evaluate_model_config(trainer):
+
+    trainer.X = np.ones((10, 3))
+    trainer.y = np.append(np.ones(5), np.zeros(5))
+
+    trainer.evaluate_model_config()
+    trainer_folder = Path(trainer.folder)
+    assert trainer_folder.is_dir()
+    assert (trainer_folder / "best_model.skops").is_file()
+    assert (trainer_folder / "model_accuracies.csv").is_file()
+    assert (trainer_folder / "model_info.json").is_file()
+
+
+def test_train_model_using_two_csvs():
+
+    metrics_path_1 = make_temp_training_csv()
+    metrics_path_2 = make_temp_training_csv()
+
+    folder = tempfile.mkdtemp()
+    metric_names = ["metric1", "metric2", "metric3"]
+
+    trainer = train_model(
+        mode="csv",
+        metrics_paths=[metrics_path_1, metrics_path_2],
+        folder=folder,
+        labels=[[0, 1, 0, 1, 0, 1, 0, 1, 0, 1], [0, 1, 0, 1, 0, 1, 0, 1, 0, 1]],
+        metric_names=metric_names,
+        imputation_strategies=["median"],
+        scaling_techniques=["standard_scaler"],
+        classifiers=["LogisticRegression"],
+        overwrite=True,
+    )
+    assert isinstance(trainer, CurationModelTrainer)
+
+
+def test_train_model():
```

Comment:
There are a lot of important parameters for this function, I think the majority are tested above. However it would be nice to check all of these are doing what is expected when this function is called (either that outputs of some test case match as expected, or they are propagated to functions that are tested elsewhere correctly using monkeypatching. This type of deeper testing is quite laborious to setup so might be too much to ask, but a benefit is you can be 100% sure your functions are doing what you expect going forward, and are not broken in some future refactoring / other code changes.